client.go

```diff
@@ -223,26 +192,97 @@ func NewClientWInterface(networkInterface string) *Client {
 
 // NewH2Client creates a RESTful client instance, forced to use HTTP2 with TLS (H2) (a.k.a. prior knowledge).
 func NewH2Client() *Client {
+	return NewH2ClientWInterface("")
+}
+
+// NewH2CClient creates a RESTful client instance, forced to use HTTP2 Cleartext (H2C).
+func NewH2CClient() *Client {
+	return NewH2CClientWInterface("")
+}
+
+// NewH2ClientWInterface creates a RESTful client instance with the http2 protocol bound to that network interface.
+// The instance has a semi-permanent transport TCP connection.
+func NewH2ClientWInterface(networkInterface string) *Client {
 	c := &Client{Kind: KindH2}
-	var rt http.RoundTripper = &h2Transport
+	var rt http.RoundTripper = getH2Transport(networkInterface)
 	if isTraced && tracer.GetOTel() {
 		rt = otelhttp.NewTransport(rt)
 	}
 	c.Client = &http.Client{Transport: rt}
 	return c
 }
 
-// NewH2CClient creates a RESTful client instance, forced to use HTTP2 Cleartext (H2C).
-func NewH2CClient() *Client {
+// NewH2ClientWInterface creates a RESTful client instance with the http2 clear text protocol bound to that network interface.
+// In other words, the http2 clear text is the http2 but without TLS handshake.
+// The instance has a semi-permanent transport TCP connection.
+func NewH2CClientWInterface(networkInterface string) *Client {
 	c := &Client{Kind: KindH2C}
-	var rt http.RoundTripper = &h2CTransport
+	var rt http.RoundTripper = getH2CTransport(networkInterface)
 	if isTraced && tracer.GetOTel() {
 		rt = otelhttp.NewTransport(rt)
 	}
 	c.Client = &http.Client{Transport: rt}
 	return c
 }
 
+func getH2Transport(iface string) *http2.Transport {
+	return &http2.Transport{
+		DialTLS: getDialTLSCallback(iface, true),
+	}
+}
+
+func getH2CTransport(iface string) *http2.Transport {
+	return &http2.Transport{
+		AllowHTTP: true,
+		DialTLS: getDialTLSCallback(iface, false),
+	}
+}
+
+func getDialTLSCallback(iface string, withTLS bool) func(string,string,*tls.Config) (net.Conn, error) {
```

Comment:
I'm pretty sure, that this is not working with H2C, because all the time  
tls.DialWithDialer
is invoked